lib/utils/class-alerts.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Automattic\VIP\Utils;
+
+class Alerts {
+	/**
+	 * The alerts service address
+	 *
+	 * @var string
+	 */
+	protected $service_address;
+
+	/**
+	 * The alerts service port
+	 *
+	 * @var string
+	 */
+	protected $service_port;
+
+	/**
+	 * Full path to the alerts service
+	 *
+	 * @var string
+	 */
+	protected $service_url;
+
+	/**
+	 * Instance of his Alerts class
+	 *
+	 * @var Alerts
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Contructor
+	 *
+	 * Set to protected to prevent direct instantiation.
+	 * Use Alerts::get_instance() to get an instance of Alerts
+	 */
+	protected function __construct() {
+		if ( ! defined( 'ALERT_SERVICE_ADDRESS' ) || ! ALERT_SERVICE_ADDRESS ) {
+			throw new Exception( 'Missing alerts service host configuration in ALERT_SERVICE_ADDRESS constant' );
```

Comment:
@chrean also mentioned the exception throwing in some other comments, but an uncaught thrown exception is a *fatal error*. That would be drastically different behavior than what we currently do, which is log an error message and return false without trying to contact the service - https://github.com/Automattic/vip-go-mu-plugins/blob/b81b054081e5eebc3b9c054c9cb058b11fe951e7/vip-helpers/vip-utils.php#L1457

This would at least break local development.

Reply:
I guess everywhere that *currently* calls the constructor is wrapped in a try/catch - but in general we should not use Exceptions and instead pass WP_Errors around, which is the "WordPress way" (so we should strive to match that for consistency with the rest of the WP ecosystem).

Reply:
This is a little tricky since constructors doesn't really return anything. Adding `return new WP_Error()` wouldn't do anything and might even cause an error. This is the primary reason I went with exceptions

Reply:
Then maybe we should not either `throw` nor `return WP_Error` inside of the constructor? Maybe only functions like `init()` should return errors? How is that normally done in WP?

Reply:
> Maybe only functions like init() should return errors? How is that normally done in WP?

I'm not sure what's the "proper" way in WP but I think using the `init()` method could work.

Reply:
I've moved the object config setup code to the `instance()` method in f49de002aae092673c9c394129520f7c3c0585c9
What do you think?

Reply:
I like it better! 👍